onmt/trainer.py

```diff
@@ -38,7 +38,10 @@ def build_trainer(opt, device_id, model, vocabs, optim, model_saver=None):
     train_loss = LossCompute.from_opts(opt, model, vocabs['tgt'])
     valid_loss = LossCompute.from_opts(opt, model, vocabs['tgt'], train=False)
 
-    scoring_preparator = ScoringPreparator(vocabs, opt)
+    scoring_preparator = ScoringPreparator(vocabs=vocabs, opt=opt)
+    validset_transforms = opt.data.get("valid", {}).get("transforms", None)
```

Comment:
do we really need this here since you added it in train_single in a previous PR ?